src/envoy/utils/utils.cc

```diff
@@ -34,6 +34,29 @@ const std::string kPerHostMetadataKey("istio");
 // Attribute field for per-host data override
 const std::string kMetadataDestinationUID("uid");
 
+bool getCertSAN(const Network::Connection* connection, bool peer,
+                std::string* principal) {
+  if (connection) {
```

Comment:
It's severe issue when when the connection is nullptr. Consider add a ENVOY_LOG?

Reply:
This is a utility function so I would prefer to add the LOG in the caller site (validateX509 in authenticator_base.cc)